### PR TITLE
cli: branch: split modules and do minor cleanup

### PR DESCRIPTION
**Pull request by yuja**
# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [x] I have added tests to cover my changes

### CODE REVIEW

File: cli/src/commands/branch/mod.rs [hunk 1]
```diff
@@ -75,15 +75,15 @@ pub fn cmd_branch(
...
-        BranchCommand::List(sub_args) => cmd_branch_list(ui, command, sub_args),
+        BranchCommand::Create(args) => cmd_branch_create(ui, command, args),
+        BranchCommand::Rename(args) => cmd_branch_rename(ui, command, args),
+        BranchCommand::Set(args) => cmd_branch_set(ui, command, args),
+        BranchCommand::Move(args) => cmd_branch_move(ui, command, args),
+        BranchCommand::Delete(args) => cmd_branch_delete(ui, command, args),
+        BranchCommand::Forget(args) => cmd_branch_forget(ui, command, args),
+        BranchCommand::Track(args) => cmd_branch_track(ui, command, args),
+        BranchCommand::Untrack(args) => cmd_branch_untrack(ui, command, args),
+        BranchCommand::List(args) => cmd_branch_list(ui, command, args),
```
Comment (martinvonz) [inline cli/src/commands/branch/mod.rs]: We seem to use `sub_args` in many of the other files. Should we change those too (can be in a separate PR)? (I wouldn't mind)

**Review by martinvonz — APPROVED** [review]
Thanks!

> cli: branch: reject non-empty branch name consistently by "set"

Drop "non-"?

File: cli/src/commands/branch/mod.rs (hunk 1 above)
Comment (yuja) [reply cli/src/commands/branch/mod.rs]: Indeed. I thought `args` is the convention, but there are both. I'll send a follow up.

**Comment by yuja** [comment]
> Drop "non-"?

Doh, bad English, fixed.
